src/mgr/ActivePyModule.h

```diff
@@ -45,12 +46,18 @@ class ActivePyModule : public PyModuleRunner
 
   std::string m_command_perms;
   const MgrSession* m_session = nullptr;
+  std::string fin_thread_name;
+public:
+  Finisher finisher; // per active module finisher to execute commands
 
 public:
   ActivePyModule(const PyModuleRef &py_module_,
       LogChannelRef clog_)
-    : PyModuleRunner(py_module_, clog_)
-  {}
+    : PyModuleRunner(py_module_, clog_),
+      fin_thread_name("m-fin-" + thread_name),
```

Comment:
what does this look like in practice? It feels strange constructing this based on the thread name.

Reply:
thread_name is actually the module name.

Reply:
```
  35   Thread 0x7f78dd20b700 (LWP 367526) "m-fin-balancer"  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407de488) at ../sysdeps/nptl/futex-internal.h:183
  37   Thread 0x7f78dc1c9700 (LWP 367528) "m-fin-crash"     futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407de748) at ../sysdeps/nptl/futex-internal.h:183
  39   Thread 0x7f78db1c7700 (LWP 367530) "m-fin-devicehea" futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407dea08) at ../sysdeps/nptl/futex-internal.h:183
  40   Thread 0x7f78da1c5700 (LWP 367532) "m-fin-iostat"    futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407decc8) at ../sysdeps/nptl/futex-internal.h:183
  41   Thread 0x7f78d91c3700 (LWP 367534) "m-fin-nfs"       futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407def88) at ../sysdeps/nptl/futex-internal.h:183
  42   Thread 0x7f78d81c1700 (LWP 367536) "m-fin-orchestra" futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407df248) at ../sysdeps/nptl/futex-internal.h:183
  44   Thread 0x7f78d71bf700 (LWP 367538) "m-fin-pg_autosc" futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407df508) at ../sysdeps/nptl/futex-internal.h:183
  46   Thread 0x7f78d61bd700 (LWP 367540) "m-fin-progress"  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407df7c8) at ../sysdeps/nptl/futex-internal.h:183
  47   Thread 0x7f78d517b700 (LWP 367542) "m-fin-status"    futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407dfa88) at ../sysdeps/nptl/futex-internal.h:183
  49   Thread 0x7f78d4179700 (LWP 367544) "m-fin-telemetry" futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x55d1407dfd48) at ../sysdeps/nptl/futex-internal.h:183

```

Reply:
But the mgr perf dump is not picking this up for some reason. I will dig in further.

```
    "finisher-Mgr": {                   <--------------------  Generic finisher thread
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 2,
            "sum": 0.114045908,
            "avgtime": 0.057022954
        }
    },
    "finisher-Mgr-fin": {         <---------------------- Per module finisher thread 1
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },
    "finisher-Mgr-fin-0x5634056695f0": {  <-------------------------- Per module finisher thread 2
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },

```

Reply:
Ok, fin_thread_name is the actual thread name displayed in gdb, finisher name is the second arg. Fixed it.

```
    "finisher-Mgr": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 2,
            "sum": 0.034990320,
            "avgtime": 0.017495160
        }
    },
    "finisher-MgrStandby": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },
    "finisher-balancer": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },
    "finisher-cmd_finisher": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 9,
            "sum": 0.000637829,
            "avgtime": 0.000070869
        }
    },
    "finisher-crash": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },
    "finisher-devicehealth": {
        "queue_len": 0,
        "complete_latency": {
            "avgcount": 0,
            "sum": 0.000000000,
            "avgtime": 0.000000000
        }
    },

```

Reply:
Very good.

> thread_name is actually the module name.

Ok, but why not:

`fin_thread_name("m-fin-" + py_module->module_name)`

I think that looks clearer.

Reply:
Done